frontend/src/templates/Field/ChildrenCompound/ChildrenCompoundField.tsx

```diff
@@ -0,0 +1,404 @@
+        align="stretch"
+        aria-describedby={`children-desc-${schema._id}`}
+        aria-labelledby={`${schema._id}-label`}
+      >
+        <VStack align="stretch" role="list">
+          {fields.map((field, currChildBodyIdx) => (
+            <ChildrenBody
+              key={`body-${currChildBodyIdx}`}
+              {...{
+                currChildBodyIdx,
+                schema,
+                fields,
+                field,
+                colorTheme,
+                remove,
+                myInfoChildrenBirthRecords,
+                isSubmitting,
+                formContext,
+              }}
+            />
+          ))}
+        </VStack>
+        {schema.allowMultiple ? (
+          <HStack>
+            <Button
+              isDisabled={isSubmitting}
+              alignSelf="start"
+              leftIcon={<BiPlus />}
+              aria-label="Add another child"
+              onClick={() => {
+                append([''])
+              }}
+            >
+              Add another child
+            </Button>
+            <Spacer />
+            <Text
+              textStyle="body-2"
+              color="secondary.400"
+            >{simplur`${numChild} child[|ren] added`}</Text>
+          </HStack>
+        ) : null}
+      </VStack>
+    </FieldContainer>
+  )
+}
+
+interface ChildrenBodyProps {
+  currChildBodyIdx: number
+  schema: ChildrenCompoundFieldSchema
+  fields: FieldArrayWithId<
+    ChildrenCompoundFieldInputs,
+    `${string}.child`,
+    'id'
+  >[]
+  field: FieldArrayWithId<ChildrenCompoundFieldInputs, `${string}.child`, 'id'>
+  colorTheme: FormColorTheme
+  remove: UseFieldArrayRemove
+  myInfoChildrenBirthRecords?: MyInfoChildData
+  isSubmitting: boolean
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  formContext: UseFormReturn<ChildrenCompoundFieldInputs, any>
+}
+
+const ChildrenBody = ({
+  currChildBodyIdx,
+  schema,
+  fields,
+  field,
+  colorTheme,
+  remove,
+  myInfoChildrenBirthRecords,
+  isSubmitting,
+  formContext,
+}: ChildrenBodyProps): JSX.Element => {
+  const { register, getValues, setValue, watch } = formContext
+
+  const childNamePath = useMemo(
+    () => `${schema._id}.child.${currChildBodyIdx}.0`,
+    [schema._id, currChildBodyIdx],
+  )
+  const { onChange: selectOnChange, ...selectRest } = register(childNamePath)
+
+  const childName = watch(childNamePath) as unknown as string
+
+  const allChilds = useMemo<string[]>(() => {
+    if (myInfoChildrenBirthRecords === undefined) {
+      return []
+    }
+    return myInfoChildrenBirthRecords[MyInfoAttribute.ChildName] ?? []
+  }, [myInfoChildrenBirthRecords])
+
+  // useCallback to re-compute names because for some reason watch doesn't
+  // work on this nested field in react-hook-form.
+  const allSelectedNames = useCallback((): string[] => {
+    // Really important to note that sometimes react-hook-form stores our "array"
+    // as a object with key=index and values=array entry.
```

Comment:
> sometimes

Any pattern to note for this? Is it due to the time of instantiation or load?

Reply:
No, it just turns up when I access it certain ways (I forgot when and how, I just noticed it does 😢 )